src/custom-element.js

```diff
@@ -591,13 +589,150 @@ function createBaseCustomElementClass(win, elementConnectedCallback) {
       );
       return readyPromise.then(() => {
         if (this.V1()) {
-          const builder = getBuilderForDoc(this.getAmpDoc());
-          builder.scheduleAsap(this);
+          const scheduler = getSchedulerForDoc(this.getAmpDoc());
+          scheduler.scheduleAsap(this);
         }
         return this.whenBuilt();
       });
     }
 
+    /**
+     * Mounts the element by calling the `BaseElement.mountCallback` method.
+     *
+     * Can only be called on a upgraded element. May only be called from
+     * scheduler.js.
+     *
+     * @return {!Promise}
+     * @final
+     * @restricted
+     */
+    mountInternal() {
+      if (this.mountPromise_) {
+        return this.mountPromise_;
+      }
+      this.mountAbortController_ = new AbortController();
+      const {signal} = this.mountAbortController_;
+      return (this.mountPromise_ = this.buildInternal()
+        .then(() => {
+          devAssert(this.V1());
+          if (signal.aborted) {
+            // Mounting has been canceled.
+            return;
+          }
+          this.setReadyStateInternal(
+            this.implClass_.usesLoading(this)
+              ? ReadyState.LOADING
+              : ReadyState.MOUNTING
+          );
+          return this.impl_.mountCallback(signal);
+        })
+        .then(() => {
+          this.mountAbortController_ = null;
+          if (signal.aborted) {
+            throw cancellation();
+          }
+          this.signals_.signal(CommonSignals.MOUNTED);
+          if (
+            this.implClass_.usesLoading(this) &&
+            this.readyState_ !== ReadyState.COMPLETE
+          ) {
+            this.setReadyStateInternal(ReadyState.LOADING);
```

Comment:
This seems a bit confusing to me. If `usesLoading`, shouldn't the mount have called `setReadyState` already? Maybe it's already errored, or something.

For dev experience, I kinda which we could ensure that any call to `setReadyState` happend if we're a loadable element.

Reply:
I think I understood what you meant and you were right. The LOADING state was already set so I change it to now simply ignore it here when `usesLoading` is set. Thanks!